### PR TITLE
HOCS-3577 Merge hocs-workflow epic/foi-mvp branch into main

### PR DESCRIPTION
**Pull request by agraham19**
HOCS-3577 Merge hocs-workflow main branch into epic/foi-mvp

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java [hunk 1]
```diff
@@ -66,9 +67,39 @@ public WorkflowService(CaseworkClient caseworkClient,
         this.userPermissionsService = userPermissionsService;
     }
 
-    public CreateCaseResponse createCase(String caseDataType, LocalDate dateReceived, List<DocumentSummary> documents, UUID userUUID, UUID fromCaseUUID) {
+    public CreateCaseResponse createCase(String caseDataType, LocalDate dateReceived, List<DocumentSummary> documents, UUID userUUID, UUID fromCaseUUID, Map<String, String> receivedData) {
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:70]: Just to confirm, the front end call was changed to pass in the map of `receivedData` by default wasn't it? We should be careful on here.

Comment (gary-newbolt) [reply src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:70]: I don't think we need to change anything, only that we make sure the corresponding change goes in to the `hocs-frontend` code.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/dto/AddressDto.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
-@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor()
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/dto/AddressDto.java:9]: Are we using this anywhere? Can the change be reverted? I can't seen an instance where we use the `new AddressDto` object instantiation anywhere... I might have missed it.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/client/infoclient/dto/StageTypeDto.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.io.Serializable;
+
+@NoArgsConstructor
+@Getter
+public class StageTypeDto implements Serializable {
+
+    //TODO: remove - used in UI
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/client/infoclient/dto/StageTypeDto.java:13]: What would be the effort required to remove these now? Given they've been highlighted here?

**Comment by gary-newbolt** [comment]
Generally looks okay to go... there are a few _possibly should do_ type changes but otherwise ok

**Review by LaurieBennett — APPROVED** [review]
All approved

**Pull request merged**